src/chaos-experiments/models/chaosExperimentsManager.js

```diff
@@ -167,3 +167,29 @@ module.exports.init = async function () {
     await reloadChaosExperiments();
     await scheduleFinishedResourcesCleanup();
 };
+
+const stopResourcesOfJobIdAndExperiment = async (jobId, kind, namespace) => {
+    try {
+        await connector.deleteAllResourcesOfKindAndJob(kind, namespace, jobId);
+    } catch (e){
+        logger.error(`Failed to get resources of kind ${kind} of jobId ${jobId}: ${e}`);
+    }
+};
+
+module.exports.stopJobExperimentsByJobId = async function(jobId) {
+    try {
+        const jobExperiments = await getChaosJobExperimentsByJobId(jobId);
+        const now = Date.now();
+        const relevantJobExperiments = jobExperiments.filter(experiment => experiment.start_time <= now);
+        const experimentIds = [...new Set(relevantJobExperiments.map(jobExperiment => jobExperiment.experiment_id))];
+        const experiments = await getChaosExperimentsByIds(experimentIds);
+        const kindNamespaceStrings = [...new Set(experiments.map(ex => `${ex.kubeObject.kind}_${ex.kubeObject.metadata.namespace}`))];
+        await Promise.all(kindNamespaceStrings.map(async(kindNamespaceString) => {
+            const splittedString = kindNamespaceString.split('_');
+            await stopResourcesOfJobIdAndExperiment(jobId, splittedString[0], splittedString[1]);
+        }
+        ));
+    } catch (e) {
+        logger.error(`Error while trying to stop job experiments for job ${jobId}`);
```

Comment:
we are overriding the error message. do we want it?

Reply:
added